utils/utils.py

```diff
@@ -136,31 +168,29 @@ def compare_dataframes(ibis_dfs, pandas_dfs, sort_cols=["id"], drop_cols=["id"])
         print("dataframes are equal")
         return
 
+    print("Fast check took {:.2f} seconds".format(timer() - t0))
+
     # comparing step
+    t0 = timer()
     for ibis_df, pandas_df in zip(ibis_dfs, pandas_dfs):
         assert ibis_df.shape == pandas_df.shape
-        for column_name in ibis_df.columns:
-            try:
-                pd.testing.assert_frame_equal(
-                    ibis_df[[column_name]],
-                    pandas_df[[column_name]],
-                    check_less_precise=2,
-                    check_dtype=False,
-                )
-            except AssertionError as assert_err:
-                if str(ibis_df.dtypes[column_name]).startswith("float"):
-                    try:
-                        current_error = get_percentage(str(assert_err))
-                        if current_error > max_error:
-                            print(
-                                f"Max acceptable difference: {max_error}%; current difference: {current_error}%"
-                            )
-                            raise assert_err
-                    # for catch exceptions from `get_percentage`
-                    except Exception:
-                        raise assert_err
-                else:
-                    raise assert_err
+        if parallel_execution:
+            from multiprocessing import Pool
+
+            pool = Pool(parallel_processes)
+            pool.map(
+                compare_columns,
+                [
+                    (ibis_df[column_name], pandas_df[column_name])
+                    for column_name in ibis_df.columns
+                ],
```

Comment:
```suggestion
                (
                    (ibis_df[column_name], pandas_df[column_name])
                    for column_name in ibis_df.columns
                ),
```
this way you'd build a `genexpr` (generator expression) which won't consume extra memory (`list` is first built and only then used while generator expressions are evaluated when needed).

Reply:
Yes, this approach is better.